lib/chef/mixin/from_file.rb

```diff
@@ -30,7 +30,7 @@ module FromFile
       # Raises an IOError if the file cannot be found, or is not readable.
       def from_file(filename)
         self.source_file = filename
-        if File.exists?(filename) && File.readable?(filename)
+        if File.exists?(filename) && File.readable?(filename) && File.file?(filename)
```

Comment:
Since we're checking .file? we can remove .exists? : 
`if File.file?(filename) && File.readable?(filename)`